From 0c9a06cf30f1058ec0fe0e549a95723119726e84 Mon Sep 17 00:00:00 2001
From: qcgg <g@quarkchain.org>
Date: Thu, 19 Jul 2018 15:08:28 -0700
Subject: [PATCH] Necessary changes to let evm tests pass

This commit reverts the QuarkChain specific changes made on EVM so that
we can run the tests to make sure we don't break other pieces of EVM.

python3 quarkchain/evm/tests/test_state.py fixtures/GeneralStateTests/
---
 quarkchain/evm/messages.py     | 8 ++++----
 quarkchain/evm/state.py        | 6 +++---
 quarkchain/evm/transactions.py | 2 +-
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/quarkchain/evm/messages.py b/quarkchain/evm/messages.py
index 6c30a0f..264ccb1 100644
--- a/quarkchain/evm/messages.py
+++ b/quarkchain/evm/messages.py
@@ -129,8 +129,8 @@ def validate_transaction(state, tx):
     if not tx.sender:  # sender is set and validated on Transaction initialization
         raise UnsignedTransaction(tx)
 
-    # assert config_fork_specific_validation(
-    #     state.config, state.block_number, tx)
+    assert config_fork_specific_validation(
+        state.config, state.block_number, tx)
 
     # (2) the transaction nonce is valid (equivalent to the
     #     sender account's current nonce);
@@ -176,7 +176,7 @@ def apply_message(state, msg=None, **kwargs):
     return bytearray_to_bytestr(data) if result else None
 
 
-def apply_transaction(state, tx: transactions.Transaction, tx_wrapper_hash):
+def apply_transaction(state, tx: transactions.Transaction, tx_wrapper_hash=b""):
     """tx_wrapper_hash is the hash for quarkchain.core.Transaction
     TODO: remove quarkchain.core.Transaction wrapper and use evm.Transaction directly
     """
@@ -399,7 +399,7 @@ def _apply_msg(ext, msg, code):
 
 
 def mk_contract_address(sender, full_shard_id, nonce):
-    return utils.sha3(rlp.encode([utils.normalize_address(sender), full_shard_id, nonce]))[12:]
+    return utils.sha3(rlp.encode([utils.normalize_address(sender), nonce]))[12:]
 
 
 def create_contract(ext, msg):
diff --git a/quarkchain/evm/state.py b/quarkchain/evm/state.py
index c4d792a..149d252 100644
--- a/quarkchain/evm/state.py
+++ b/quarkchain/evm/state.py
@@ -56,15 +56,15 @@ class Account(rlp.Serializable):
         ('balance', big_endian_int),
         ('storage', trie_root),
         ('code_hash', hash32),
-        ('full_shard_id', BigEndianInt(4)),
     ]
 
-    def __init__(self, nonce, balance, storage, code_hash, full_shard_id, env, address, db=None):
+    def __init__(self, nonce, balance, storage, code_hash, full_shard_id=0, env=None, address=None, db=None):
         self.db = env.db if db is None else db
         assert isinstance(db, Db)
         self.env = env
         self.address = address
-        super(Account, self).__init__(nonce, balance, storage, code_hash, full_shard_id)
+        self.full_shard_id = full_shard_id
+        super(Account, self).__init__(nonce, balance, storage, code_hash)
         self.storage_cache = {}
         self.storage_trie = SecureTrie(Trie(self.db))
         self.storage_trie.root_hash = self.storage
diff --git a/quarkchain/evm/transactions.py b/quarkchain/evm/transactions.py
index a2929c5..415a8a3 100644
--- a/quarkchain/evm/transactions.py
+++ b/quarkchain/evm/transactions.py
@@ -72,7 +72,7 @@ class Transaction(rlp.Serializable):
     def __init__(self, nonce, gasprice, startgas, to, value, data,
                  v=0, r=0, s=0, fromFullShardId=0, toFullShardId=0, networkId=1, version=0):
         self.data = None
-        self.shardSize = 0
+        self.shardSize = 1
 
         to = utils.normalize_address(to, allow_blank=True)
 
-- 
2.15.2 (Apple Git-101.1)

